### PR TITLE
[EMCAL-548] Using raw fitter to convert digits to cells

### PR DESCRIPTION
**Pull request by hahassan7**
As the digitiser now digitises hits with time information the digits->cell converter must also do the raw fitting in the same way as the raw->cell converter. The steps include:
    Separation of ALTRO bunches and adding the time stamps
    filtering of all labels belonging to digits in the same ALTRO bunch
    Performing the raw fit on the ALTRO bunch in the same way as in data

The Handling of MC labels is not done yet.

### CODE REVIEW

File: Detectors/EMCAL/workflow/include/EMCALWorkflow/CellConverterSpec.h [hunk 1]
```diff
@@ -15,13 +15,39 @@
...
+
+struct ChannelData {
+  o2::emcal::ChannelType_t mChanType;
+  std::vector<o2::emcal::Bunch> mChannelsBunches;
+};
+
+struct SRUBunchContainer {
+  int mSRUid;
+  std::map<int, ChannelData> mChannelsData;
+};
```
Comment (mfasDa) [inline Detectors/EMCAL/workflow/include/EMCALWorkflow/CellConverterSpec.h]: I think they are already defined somewhere else, probably in the raw writer, therefore we would get a duplication of symbols. I guess it is code duplication so better extract them in a separate file and use them for both.

**Review by mfasDa — CHANGES_REQUESTED** [review]
AltroBunch, ChannelDigits, ChannelData and SRUDigitContainer are commonly used in RawWriter and CellConverter, thus they need to be extracted from the two classes in a separate header in order to prevent symbol duplication.

Otherwise fine

**Comment by ktf** [comment]
@pzhristov there seems to still be some issues with LLVM / Clang.

```
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createNewGVNPass()'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::Type::getInt128Ty(llvm::LLVMContext&)'
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createTargetTransformInfoWrapperPass(llvm::TargetIRAnalysis)'
collect2: error: ld returned 1 exit status
[716/3372] Building CXX object Framework/Utils/CMakeFiles/O2test-dplutils-DPLRawPageSequencer.dir/test/test_DPLRawPageSequencer.cxx.o
[717/3372] Building CXX object Framework/Core/CMakeFiles/O2test-framework-TableBuilder.dir/test/test_TableBuilder.cxx.o
```

**Comment by pzhristov** [comment]
> @pzhristov there seems to still be some issues with LLVM / Clang.
> 
> ```
> /sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createNewGVNPass()'
> /sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::Type::getInt128Ty(llvm::LLVMContext&)'
> /sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-1/bin/../lib/gcc/x86_64-unknown-linux-gnu/10.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /sw/slc8_x86-64/arrow/v5.0.0-alice2-2/lib/libgandiva.so.500.0.0: undefined reference to `llvm::createTargetTransformInfoWrapperPass(llvm::TargetIRAnalysis)'
> collect2: error: ld returned 1 exit status
> [716/3372] Building CXX object Framework/Utils/CMakeFiles/O2test-dplutils-DPLRawPageSequencer.dir/test/test_DPLRawPageSequencer.cxx.o
> [717/3372] Building CXX object Framework/Core/CMakeFiles/O2test-framework-TableBuilder.dir/test/test_TableBuilder.cxx.o
> ```

Strange, this reappeared suddenly. During the last days we haven't seen it.

**Comment by pzhristov** [comment]
@ktf Can we check if arrow was compiled with -DLLVM_ROOT=${CLANG_ROOT}?

**Comment by davidrohr** [comment]
the CI is now green, what is the status of this PR?

**Comment by hahassan7** [comment]
This pull request can be merged now.

**Review by mfasDa — APPROVED** [review]
Looks good now, PR can be merged.

**Comment by mfasDa** [comment]
Would also be in favour of merging ...

**Review by shahor02 — APPROVED** [review]

**Pull request merged**